syft/execution/placeholder.py

```diff
@@ -20,6 +21,10 @@ def __init__(self, owner=None, id=None, tags: set = None, description: str = Non
         """
         super().__init__(id=id, owner=owner, tags=tags, description=description)
 
+        # TODO currently we only use ObjectId on placeholder but if we use them, we
+        # should use them everywhere
```

Comment:
I was hoping for a quick extension to all objects because I don't want to have Placeholder as a different object... But I can change the naming for now and change it back when we extend it to other objects.
What do you think?

Reply:
I think it's still beneficial to know what type the id goes with somehow. Using a separate class allows us to put that information into type annotations, which is kinda nice. On the other hand, a proliferation of tiny id classes might not be great. 🤷‍♂ 

Could there be a base `Id` class that defines the fields/behaviors and then a bunch of sub-classes that only exist to capture/enforce id types?

Reply:
Hum, I don't think `Ids` will have lots of functionalities so I'm not a fan of having a Id class for each PySyft object...

Reply:
I think we could kick this can down the road by calling it `PlaceholderId` for now. Since a one-to-one correspondence is required between Python classes and Protobuf schemas, that would require adding a schema for `PlaceholderId` though.

It's not perfect and we may just refactor it out later, but it keeps the scope of this PR limited and make it easier to think through the potential implications.